azuread/resource_domain_service.go

```diff
@@ -0,0 +1,492 @@
+							Optional: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								string(aad.NtlmV1Enabled),
+								string(aad.NtlmV1Disabled),
+							}, false),
+							Default: string(aad.NtlmV1Enabled),
+						},
+						"sync_ntlm_passwords": {
+							Type:     schema.TypeString,
+							Optional: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								string(aad.SyncNtlmPasswordsEnabled),
+								string(aad.SyncNtlmPasswordsDisabled),
+							}, false),
+							Default: string(aad.SyncNtlmPasswordsEnabled),
+						},
+						"tls_v1": {
+							Type:     schema.TypeString,
+							Optional: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								string(aad.TLSV1Enabled),
+								string(aad.TLSV1Disabled),
+							}, false),
+							Default: string(aad.TLSV1Enabled),
+						},
+					},
+				},
+			},
+
+			"filtered_sync": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					string(aad.FilteredSyncEnabled),
+					string(aad.FilteredSyncDisabled),
+				}, false),
+				Default: string(aad.FilteredSyncEnabled),
+			},
+
+			"ldaps_settings": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"external_access": {
+							Type:     schema.TypeString,
+							Optional: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								string(aad.Enabled),
+								string(aad.Disabled),
+							}, false),
+							Default: string(aad.Enabled),
+						},
+						"ldaps": {
+							Type:     schema.TypeString,
+							Optional: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								string(aad.LdapsEnabled),
+								string(aad.LdapsDisabled),
+							}, false),
+							Default: string(aad.LdapsEnabled),
+						},
+						"pfx_certificate": {
+							Type:         schema.TypeString,
+							Optional:     true,
+							Sensitive:    true,
+							ValidateFunc: validate.Base64String(),
+						},
+						"pfx_certificate_password": {
+							Type:      schema.TypeString,
+							Optional:  true,
+							Sensitive: true,
+						},
+						"external_access_ip_address": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+
+			"notification_settings": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"additional_recipients": {
+							Type:     schema.TypeList,
+							Optional: true,
+							Elem: &schema.Schema{
+								Type: schema.TypeString,
+							},
+						},
+						"notify_dc_admins": {
+							Type:     schema.TypeString,
+							Optional: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								string(aad.NotifyDcAdminsEnabled),
+								string(aad.NotifyDcAdminsDisabled),
+							}, false),
+							Default: string(aad.NotifyDcAdminsEnabled),
+						},
```

Comment:
As this is a boolean on/off value could we use a bool and pass the strings in to the API behind the scenes?